ballerina/channel_csv_io.bal

```diff
@@ -40,34 +40,80 @@ Error {
+    if (content is string[][]){
     foreach string[] r in content {
         var writeResult = csvChannel.write(r);
         if writeResult is Error {
             check csvChannel.close();
             return writeResult;
+            }
+        }
+    }
+    else if(content is map<anydata>[]) {
+        map<anydata> first = content[0];
+        string[] keys=first.keys();
+        //check csvChannel.write(keys);
+        foreach map<anydata> r in content{
+            string[] temp=[];
+            int count=0;
+            
+            foreach string t in keys{
+                temp[count]=r[t].toString();
+                count+=1;
+            }
+            var writeResult = csvChannel.write(temp);
+            if writeResult is Error {
+                check csvChannel.close();
+                return writeResult;
+        }
+
         }
     }
     check csvChannel.close();
     return;
 }
 
-isolated function channelWriteCsvFromStream(WritableChannel writableChannel, stream<string[], Error?> csvStream) returns
+isolated function channelWriteCsvFromStream(WritableChannel writableChannel, stream<string[]|map<anydata>, Error?> csvStream) returns
 Error? {
     WritableCSVChannel csvChannel = check getWritableCSVChannel(writableChannel);
-    do {
-        record {|string[] value;|}|Error? csvRecord = csvStream.next();
-        while csvRecord is record {|string[] value;|} {
-            check csvChannel.write(csvRecord.value);
-            csvRecord = csvStream.next();
+    if (csvStream is stream<string[], Error?>){
+        record {|string[] value;|}|Error? csvRecordString = csvStream.next();
+        do {
+            while csvRecordString is record {|string[] value;|} {
+                check csvChannel.write(csvRecordString.value);
+                csvRecordString = csvStream.next();
+            }
+            check csvChannel.close();
+        } on fail Error err {
+            check csvChannel.close();
+            return err;
+        }
+    } else if (csvStream is stream<map<anydata>, Error?>){
+        record {|map<anydata> value;|}? csvRecordMap = check csvStream.next();
+        string[] keys=[];
+        if csvRecordMap !is () {
+            keys = csvRecordMap["value"].keys();
         }
-        check csvChannel.close();
-    } on fail Error err {
-        check csvChannel.close();
-        return err;
+        //check csvChannel.write(keys);
+        do {
+            while csvRecordMap is record {|map<anydata> value;|} {
+                string[] temp=[];
+                int count=0;
+                foreach any t in keys{
+                    temp[count]=csvRecordMap.value[t].toString();
+                    count+=1;
+                }
+                check csvChannel.write(temp);                
+                csvRecordMap = check csvStream.next();
+            }
+        } on fail Error err {
+            check csvChannel.close();
+            return err;
+        }   
```

Comment:
Same as above

Reply:
simplified the logic, but cannot use for each here as the termination type contains Error